compiler/rustc_const_eval/src/const_eval/mod.rs

```diff
@@ -164,5 +165,31 @@ pub(crate) fn deref_mir_constant<'tcx>(
         },
     };
 
-    mir::ConstantKind::Val(op_to_const(&ecx, &mplace.into()), ty)
+    mir::ConstantKind::Val(op_to_const(&mut ecx, &mplace.into()), ty)
+}
+
+pub fn slice_for_alloc<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    alloc: ConstAllocation<'tcx>,
+) -> ConstAllocation<'tcx> {
+    let addr = tcx.create_memory_alloc(alloc);
+    let ptr_size = tcx.data_layout.pointer_size;
+
+    let a = Scalar::Ptr(Pointer::new(addr, Size::ZERO), ptr_size.bytes().try_into().unwrap());
```

Comment:
You can use `Scalar::from_pointer` here, and `addr.into()`.
(I am a bit surprised that this works without wrapping `addr` into an `Option`.)